libraries/AP_Frsky_Telem/mavlite.h

```diff
@@ -0,0 +1,74 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+#include <stdint.h>
+#define MAVLITE_MAX_PAYLOAD_LEN 31 //7 float params + cmd_id + options
+
+#ifndef HAL_WITH_FRSKY_TELEM_BIDIRECTIONAL
+#define HAL_WITH_FRSKY_TELEM_BIDIRECTIONAL 1
+#endif
+
+#if HAL_WITH_FRSKY_TELEM_BIDIRECTIONAL
+
+namespace MAVLITE
+{
+typedef struct {
+    uint8_t msgid = 0;      // ID of message in payload
+    uint8_t len = 0;        // Length of payload
+    uint8_t payload[MAVLITE_MAX_PAYLOAD_LEN];
+    int16_t checksum = 0;   // sent at end of packet
+} mavlite_message_t;
+
+typedef enum : uint8_t {
```

Comment:
`enum class ParseState`, please.

Don't  forget to hack off the extra bits off the entries after that :-)

Reply:
ok